mock/src/lib.rs

```diff
@@ -68,3 +68,81 @@ pub fn eth(x: u64) -> Word {
 pub fn gwei(x: u64) -> Word {
     Word::from(x) * Word::from(10u64.pow(9))
 }
+
+/// Struct that holds the parameters for generating mock EVM bytecode.
+pub struct MockBytecodeParams {
+    pub address: Address,
+    pub pushdata: Vec<u8>,
+    pub return_data_offset: usize,
+    pub return_data_size: usize,
+    pub call_data_length: usize,
+    pub call_data_offset: usize,
+    pub gas: u64,
+}
+
+/// Set default parameters for MockBytecodeParams
+impl Default for MockBytecodeParams {
+    fn default() -> Self {
+        MockBytecodeParams {
+            address: address!("0x0000000000000000000000000000000000000000"),
+            pushdata: Vec::new(),
+            return_data_offset: 0x00usize,
+            return_data_size: 0x00usize,
+            call_data_length: 0x00usize,
+            call_data_offset: 0x00usize,
+            gas: 0x1_0000u64,
+        }
+    }
+}
+
+/// Generates mock EVM bytecode that performs a simple contract call
+pub fn generate_mock_bytecode(params: MockBytecodeParams) -> Vec<u8> {
+    let bytecode = bytecode! {
+        // populate memory in the context.
+        PUSH32(Word::from_big_endian(&params.pushdata))
+        PUSH1(0x00) // offset
+        MSTORE
+        // call address
+        PUSH32(params.return_data_offset) // retLength
+        PUSH32(params.return_data_size) // retOffset
+        PUSH32(params.call_data_length) // argsLength
+        PUSH32(params.call_data_offset) // argsOffset
+        PUSH1(0x00) // value
+        PUSH32(params.address.to_word()) // address
+        PUSH32(params.gas) // gas
+        CALL
+        STOP
+    };
+    bytecode.to_vec()
+}
+
+/// Generates mock bytecode with an additional instruction
+/// that is added before the STOP instruction.
+pub fn generate_mock_bytecode_with_instruction(
+    params: MockBytecodeParams,
+    instruction: Vec<u8>,
+) -> Vec<u8> {
+    let mut bytecode = generate_mock_bytecode(params);
+    let last_index = bytecode.len() - 1;
+    bytecode.splice(last_index..last_index, instruction);
+    bytecode
+}
```

Comment:
Maybe cleaner if you add `instruction` to `MockBytecodeParams` that can be set, then you only need to modify `generate_mock_bytecode` to insert `instruction` between CALL and STOP and there's no need for a separate function.

Reply:
Updated. Add `instructions_before_stop` to `MockBytecodeParams`